File: solutions/0496-next-greater-element-i/solution.py
class Solution(object):
    def nextGreaterElement(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        m=max(nums2)
        for i,j in enumerate(nums1):
            if j==m:
                nums1[i]=-1
            else:
                for k in range(nums2.index(j),len(nums2)):
                    if nums2[k]>j:
                        nums1[i]=nums2[k]
                        break;
                else:
                    nums1[i]=-1
        return nums1;
                

File: solutions/0496-next-greater-element-i/solution.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        map<int,int>map;
        stack<int>stack;

        int n2=nums2.size(),n1=nums1.size();
        vector<int> res(n1,-1);

        for(int i=n2-1;i>=0;i--)
        {
            while(!stack.empty() && stack.top()<=nums2[i])
                stack.pop();
            if(stack.empty())
                map[nums2[i]]=-1;
            else
                map[nums2[i]]=stack.top();
            stack.push(nums2[i]);
        }
        for(int i=0;i<n1;i++)
            res[i]=map[nums1[i]];
        return res;
    }
};
